src/expint.jl

```diff
@@ -178,25 +191,23 @@ function En_cf_nogamma(ν::Number, z::Number, n::Int=1000)
         conv && i > 4 && break
 
         # rescale
-        if fastabs(A) > scale
+        if max(abs(real(A)), abs(imag(A))) > scale
             A     /= scale
             Aprev /= scale
             B     /= scale
             Bprev /= scale
         end
     end
 
-    exppart = exp(-z)
-    if isinf(exppart)
-        return exp(-z + log(A) - log(B)), iters
-    else
-        cfpart = A/B
-        return cfpart * exppart, iters
-    end
+    return A/B, iters
 end
 
 # Calculate Γ(1 - ν) * z^(ν-1) safely
-En_safe_gamma_term(ν::Number, z::Number) = exp((ν - 1)*log(z) + loggamma(1 - oftype(z, ν)))
+function En_safe_gamma_term(ν::Number, z::Number)
+    arg = 1 - oftype(z, ν)
+    lgamma, lgammasign = arg isa Real ? logabsgamma(arg) : (loggamma(arg), 1)
+    return lgammasign * exp((ν - 1)*log(z) + lgamma)
+end
 En_safe_gamma_term(ν::Integer, z::Real) = (z ≥ 0 || isodd(ν) ? 1 : -1) * exp((ν - 1)*log(abs(z)) + loggamma(1 - oftype(z, ν)))
```

Comment:
Need to merge this into your method above — otherwise if `z isa Real` and `z < 0` then `log(z)` throws.

Reply:
Nevermind, if `ν` is an integer then I think `gamma(1 - oftype(z, ν))` will never be negative?